limitador/src/storage/redis/redis_async.rs

```diff
@@ -145,14 +145,14 @@ impl AsyncCounterStorage for AsyncRedisStorage {
 }
 
 impl AsyncRedisStorage {
-    pub async fn new(redis_url: &str) -> Self {
-        Self {
+    pub async fn new(redis_url: &str) -> Result<Self, RedisError> {
+        let info = ConnectionInfo::from_str(redis_url)?;
+        Ok(Self {
             conn_manager: ConnectionManager::new(
-                redis::Client::open(ConnectionInfo::from_str(redis_url).unwrap()).unwrap(),
+                redis::Client::open(info).expect("Somehow couldn't create Redis client!"),
```

Comment:
Am I right expecting that the function `ConnectionManager::new` will return error when the connection cannot be established?  Because, if it is `redis::Client::open(info)` it will panic and `AsyncRedisStorage::new` will not return error, but panic

Reply:
So that one is really weird: 

```rust
    /// Connects to a redis server and returns a client.  This does not
    /// actually open a connection yet but it does perform some basic
    /// checks on the URL that might make the operation fail.
    pub fn open<T: IntoConnectionInfo>(params: T) -> RedisResult<Client> {
        Ok(Client {
            connection_info: params.into_connection_info()?,
        })
    }
```

and 

```rust
impl IntoConnectionInfo for ConnectionInfo {
    fn into_connection_info(self) -> RedisResult<ConnectionInfo> {
        Ok(self)
    }
}
```

I think this is a less optimal API into the client, but what we do, create the `ConnectionInfo` prior to `open` (which doesn't open), can't ever fail. I added the `expect` there, in case this API's implementation changes in the future.